task/yq/0.4/yq.yaml

```diff
@@ -0,0 +1,35 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: yq
+  labels:
+    app.kubernetes.io/version: "0.4"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Developer Tools
+    tekton.dev/tags: yq
+    tekton.dev/displayName: "YQ"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: >-
+    This task can be used to replace fields in YAML files. For example for altering helm charts on GitOps repos.
+  workspaces:
+    - name: source
+      description: A workspace that contains the file which needs to be altered.
+  params:
+    - name: SCRIPT
+      type: string
+      description: The yq script to execute. Can be multiple lines for complex tasks.
```

Comment:
@frerikandriessen I can see that a few existing params have been removed and in replace we have a new param `SCRIPT`. This can be a huge breaking change for existing users whenever they try to bump the version. How about we keep those params along with this new param `SCRIPT`? We can have a default value of `SCRIPT` param  to contain the script which was present earlier. This will not break existing user's experience. WDYT?

Reply:
@vinamra28

Reply:
@frerikandriessen tekton doesn't has this as of now. Probably in the param description we can say it as deprecated.

Reply:
@vinamra28 because I needed to get the params in a backwards compatible way I've solved it like this, is that also what you intended? Let me know

Reply:
@frerikandriessen yes. I hope this workaround should work 😅